examples/rust/tcp-ping-pong.rs

```diff
@@ -334,13 +350,18 @@ impl TcpClient {
 
             // Push data.
             {
-                let sga: demi_sgarray_t = mksga(&mut self.libos, BUFFER_SIZE, fill_char)?;
-                if let Err(e) = push_and_wait(&mut self.libos, self.sockqd, &sga) {
-                    freesga(&mut self.libos, sga);
+                self.sga = Some(mksga(&mut self.libos, BUFFER_SIZE, fill_char)?);
+                if let Err(e) = push_and_wait(
+                    &mut self.libos,
+                    self.sockqd,
+                    &self.sga.expect("should be a valid sgarray"),
+                ) {
+                    freesga(&mut self.libos, self.sga.expect("should be a valid sgarray"));
                     anyhow::bail!("push and wait failed: {:?}", e);
                 }
-                if let Err(e) = self.libos.sgafree(sga) {
-                    anyhow::bail!("failed to release scatter-gather array: {:?}", e)
+                match self.libos.sgafree(self.sga.expect("should be a valid sgarray")) {
```

Comment:
You can use the take function here, eg go back to the if statement but use take() to set the Option to None.

Reply:
Done.